flang/lib/Lower/Bridge.cpp

```diff
@@ -2009,6 +2010,8 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     else
       genFIRProcedureExit(funit, funit.getSubprogramSymbol());
 
+    // immediately throw away any dead code just created
+    mlir::simplifyRegions({builder->getRegion()});
```

Comment:
This looks like it should help.

Reply:
Yes. It fixed a bunch of assertion crashes in LAPACK.